clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/AbstractStorageEngine.java

```diff
@@ -389,66 +392,137 @@ public synchronized void endBatchWrite(StoragePartitionConfig storagePartitionCo
     }
   }
 
+  private void executeWithSafeGuard(int partitionId, Runnable runnable) {
+    executeWithSafeGuard(partitionId, () -> {
+      runnable.run();
+      return null;
+    });
+  }
+
+  private <T> T executeWithSafeGuard(int partitionId, Callable<T> callable) {
+    ReadWriteLock readWriteLock = getRWLockForPartitionOrThrow(partitionId);
+    readWriteLock.readLock().lock();
+    try {
+      return callable.call();
+    } catch (Exception e) {
+      if (e instanceof VeniceException) {
```

Comment:
Hmm, can this be simplified to new VeniceException(e) only?

Reply:
This logic was trying to make the existing test work and it is not necessary to wrap up a VeniceException with a new VeniceException.